source/modulo_components/modulo_components/component_interface.py

```diff
@@ -512,21 +513,23 @@ def add_tf_listener(self):
         else:
             self.get_logger().error("TF buffer and listener already exist.")
 
-    def send_transform(self, transform: sr.CartesianPose):
+    def send_transform(self, transform: Union[sr.CartesianPose, List[sr.CartesianPose]]):
         """
-        Send a transform to TF.
+        Send a transform or a list of transforms to TF.
 
-        :param transform: The transform to send
+        :param transform: The transform or list of transforms to send
         """
-        self.__publish_transform(transform)
+        list_transforms = transform if isinstance(transform, Iterable) else [transform]
+        self.__publish_transform(list_transforms)
 
-    def send_static_transform(self, transform: sr.CartesianPose):
+    def send_static_transform(self, transform: Union[sr.CartesianPose, List[sr.CartesianPose]]):
         """
-        Send a static transform to TF.
+        Send a static transform or a list of static transforms to TF.
 
-        :param transform: The transform to send
+        :param transform:  The transform or list of transforms to send
         """
-        self.__publish_transform(transform, static=True)
+        list_transforms = transform if isinstance(transform, Iterable) else [transform]
+        self.__publish_transform(list_transforms, static=True)
```

Comment:
If the type of `transform` is either `sr.CartesianPose` or `Iterable[sr.CartesianPose]`, then you can just do
```suggestion
        self.__publish_transform(list(transform))
```
The `list` constructor will iterate over the object to create a list, even if the iterable is already a list

Reply:
This was wrong, and I was wrong, and I am sorry. The list constructor takes iterables, yes, but _only_ iterables, so it fails for a single transform. The `isinstance` check is also nicer because it doesn't need to iterate and copy construct, so it doesn't get slower for bigger lists.